File: clients/php/tests/peek.php
<?php
use RedisPriorityQueue as RQP;

// Requires
require_once(__DIR__.'/../utils/redis_instance.php');
require_once(__DIR__.'/../utils/const.php');
require_once(__DIR__.'/../lib/Lua.php');
require_once(__DIR__.'/../lib/Queue.php');

// Create Queue instance with Redis connection
$q = new RQP\Queue($r->getConnection());

// Set queue name
$q->setQueueName(QUEUE_NAME);

// Run
echo "* peek (one):\n";
$res = $q->peek();
echo json_encode($res)."\n";

// Run
echo "* peek (one, descending):\n";
$res = $q->peek('desc');
echo json_encode($res)."\n";

// Run
echo "* peek (many):\n";
$res = $q->peek('asc', MULTIPLE_ITEMS_COUNT);
echo json_encode($res)."\n";

File: clients/python/peek.py
#!/usr/bin/env python3

# Packages
import redis
from lib.RpqQueue import RpqQueue
from lib.RpqLua import RpqLua
from utils.redis_vars import *

def listItems(items):
    '''Helper to print items'''
    for item in items:
        print (item);

# Redis instance
# !!! set Redis credentials into utils/redis_vars.py
r = redis.StrictRedis(host=rHost, port=rPort, db=rDbnum, password=rAuth)

# Load LUA Script
RpqLua = RpqLua(r, '../../src/redis-priority-queue.lua');
queue = RpqLua.register();

# RpqQueue instance
RpqQueue = RpqQueue(queue);

# Set queue name
RpqQueue.setqueueName('simple_queue')

# Peek in a queue

print ('* peek (one):')
item = RpqQueue.peekOne()
if item:
    print (item);
else:
    print ('Queue is empty');

print ('* peek (one, ascending):')
item = RpqQueue.peekOne('asc')
if item:
    print (item);
else:
    print ('Queue is empty');

print ('* peek (many):')
items = RpqQueue.peekMany('desc', 5)
if items:
    listItems(items)
else:
    print ('Queue is empty');
